presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.io.LocationProvider;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static com.facebook.presto.hive.HiveMetadata.TABLE_COMMENT;
+import static com.facebook.presto.iceberg.IcebergErrorCode.ICEBERG_INVALID_SNAPSHOT_ID;
+import static com.facebook.presto.iceberg.TypeConverter.toPrestoType;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Lists.reverse;
+import static com.google.common.collect.Streams.stream;
+import static java.lang.String.format;
+import static org.apache.iceberg.BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE;
+import static org.apache.iceberg.BaseMetastoreTableOperations.TABLE_TYPE_PROP;
+import static org.apache.iceberg.LocationProviders.locationsFor;
+import static org.apache.iceberg.TableProperties.DEFAULT_FILE_FORMAT;
+import static org.apache.iceberg.TableProperties.DEFAULT_FILE_FORMAT_DEFAULT;
+import static org.apache.iceberg.TableProperties.WRITE_LOCATION_PROVIDER_IMPL;
+
+final class IcebergUtil
+{
+    private static final Pattern SIMPLE_NAME = Pattern.compile("[a-z][a-z0-9]*");
+
+    private IcebergUtil() {}
+
+    public static boolean isIcebergTable(com.facebook.presto.hive.metastore.Table table)
+    {
+        return ICEBERG_TABLE_TYPE_VALUE.equalsIgnoreCase(table.getParameters().get(TABLE_TYPE_PROP));
+    }
+
+    public static Table getIcebergTable(ExtendedHiveMetastore metastore, HdfsEnvironment hdfsEnvironment, ConnectorSession session, SchemaTableName table)
+    {
+        HdfsContext hdfsContext = new HdfsContext(session, table.getSchemaName(), table.getTableName());
+        TableOperations operations = new HiveTableOperations(
+                metastore,
+                new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), Optional.empty()),
+                hdfsEnvironment,
+                hdfsContext,
+                table.getSchemaName(),
+                table.getTableName());
+        return new BaseTable(operations, quotedTableName(table));
+    }
+
+    public static long resolveSnapshotId(Table table, long snapshotId)
+    {
+        if (table.snapshot(snapshotId) != null) {
+            return snapshotId;
+        }
+
+        return reverse(table.history()).stream()
+                .filter(entry -> entry.timestampMillis() <= snapshotId)
+                .map(HistoryEntry::snapshotId)
+                .findFirst()
+                .orElseThrow(() -> new PrestoException(ICEBERG_INVALID_SNAPSHOT_ID, format("Invalid snapshot [%s] for table: %s", snapshotId, table)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: snapshotId. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 93. Consider formatting this data with &#39;%d&#39; instead.